terraform/justfile

```diff
@@ -0,0 +1,68 @@
+set export := true # Just variables are exported into environment variables
+
+[private]
+default:
+    just --list
+
+[private]
+stop-local-registry:
+    docker stop registry || true
+    docker rm registry || true
+
+[private]
+start-local-registry: (stop-local-registry)
+    docker run -d -p 5000:5000 --name registry registry:2.7
+
+[private]
+build-docker-image: (start-local-registry)
+    cd test && docker buildx build -f ./Dockerfile -t localhost:5000/temporal-worker:test-terraform .
+
+
+[private]
+initialize:
+    #!/usr/bin/bash
+    if [ ! -d ".terraform" ]; then
+        terraform init
+    fi
+
+[private]
+destroy-model:
+    juju destroy-model --no-prompt --destroy-storage terraform || true
+
+[private]
+add-model: (destroy-model)
+    juju add-model terraform
+
+# Run terraform format
+fmt: (initialize)
+    terraform fmt
+
+# Run terraform validate
+validate: (initialize)
+    terraform validate
+
+# Run tflint
+lint:
+    tflint
+
+# Apply the terraform plan with the provided variables file
+apply variables_file: (initialize)
+    terraform apply -auto-approve -var-file="${variables_file}"
+
+# Destroy the terraform plan with the provided variables file
+destroy variables_file:
+    terraform destroy -auto-approve -var-file="${variables_file}"
+
+    just destroy-model
+
+test: (build-docker-image)
+    #!/usr/bin/bash
+    set -euxo pipefail
+
+    trap 'just destroy "./test/test.tfvars"' EXIT
+
+    just add-model
+
+    just apply "./test/test.tfvars"
+
+    juju wait-for model terraform --query="forEach(units, unit => (unit.workload-status == 'blocked'))"
```

Comment:
Is it possible to check that it is blocked because of an expected reason?

Reply:
checking workload-message as well in 28121b232bdb1ac9731f7be3c2fb9368602b10df